apps/teams-test-app/src/components/PagesTabsAPIs.tsx

```diff
@@ -13,9 +14,24 @@ const NavigateToTab = (): React.ReactElement =>
           throw new Error('tabName is required');
         }
       },
-      submit: async input => {
-        await pages.tabs.navigateToTab(input);
-        return 'Completed';
+      submit: {
+        withPromise: async input => {
+          await pages.tabs.navigateToTab(input);
+          return 'Completed';
+        },
+        withCallback: (input, setResult) => {
+          const onComplete = (status: boolean, reason?: string): void => {
+            if (!status) {
+              if (reason) {
+                setResult(JSON.stringify(reason));
+              }
+            } else {
+              setResult('Completed');
+            }
+          };
+          navigateToTab(input, onComplete);
+          return 'navigateToTab()' + noHostSdkMsg;
```

Comment:
The more APIs you move over the more I'm questioning the initial design for `withCallback`. Maybe we should change it not to require a `string` to be returned? `ApiWithTextInput` component already seeds the UI with `noHostSdkMsg` before calling into any of the delegates, so this feels a bit redundant. What do you think? cc: @jadahiya-MSFT

Reply:
This feels a bit redundant to me as well. I will make another PR to update the `withCallback` util.